README.md

```diff
@@ -17,11 +38,107 @@ Corpus][corpus], as described by Thorsten Brants and Alex Franz, and [distribute
 education and research"**, so for any other usage you should acquire a different data set.
 
-For the microbenchmark included in this repository, instant-segment is ~17x faster than
+For the microbenchmark included in this repository, Instant Segment is ~17x faster than
 the Python implementation. Further optimizations are planned -- see the [issues][issues].
 The API has been carefully constructed so that multiple segmentations can share
 the underlying state to allow parallel usage.
 
+## Installing
+
+### Python **(>= 3.9)**
+
+```sh
+pip install instant-segment
+```
+
+### Rust
+
+```toml
+[dependencies]
+instant-segment = "*"
+```
+
+## Using
+
+Instant Segment works by segmenting a string into words by selecting the splits with the highest probability given a corpus of words and their occurances.
+
+For instance, provided that `choose` and `spain` occur more frequently than `chooses` and `pain`, Instant Segment can help you split the string `choosespain.com` into [`ChooseSpain.com`](https://instantdomainsearch.com/search/sale?q=choosespain) which more likely matches user intent.
+
+```python
+import instant_segment
+
+
+def main():
+    unigrams = []
+    unigrams.append(("choose", 50))
+    unigrams.append(("chooses", 10))
+    unigrams.append(("spain", 50))
+    unigrams.append(("pain", 10))
+
+    bigrams = []
+    bigrams.append((("choose", "spain"), 10))
+    bigrams.append((("chooses", "pain"), 10))
+
+    segmenter = instant_segment.Segmenter(iter(unigrams), iter(bigrams))
+    search = instant_segment.Search()
+    segmenter.segment("choosespain", search)
+    print([word for word in search])
+
+
+if __name__ == "__main__":
+    main()
+
+```
+
+```rust
+use instant_segment::{Search, Segmenter};
+use std::collections::HashMap;
+
+fn main() {
+    let mut unigrams = HashMap::default();
+
+    unigrams.insert("choose".into(), 50 as f64);
+    unigrams.insert("chooses".into(), 10 as f64);
+
+    unigrams.insert("spain".into(), 50 as f64);
+    unigrams.insert("pain".into(), 10 as f64);
+
+    let mut bigrams = HashMap::default();
+
+    bigrams.insert(("choose".into(), "spain".into()), 10 as f64);
+    bigrams.insert(("chooses".into(), "pain".into()), 10 as f64);
+
+    let segmenter = Segmenter::from_maps(unigrams, bigrams);
+    let mut search = Search::default();
+
+    let words = segmenter.segment("choosespain", &mut search).unwrap();
+
+    println!("{:?}", words.collect::<Vec<&str>>())
+}
+```
+
+```
+['choose', 'spain']
+```
+
+Play with the examples above to see that different numbers of occurances will influence the results
+
+The example above is succinct but, in practice, you will want to load these words and occurances from a corpus of data like the ones we provide [here](./data). Check out [the](./instant-segment/instant-segment-py/test/test.py) [tests](./instant-segment/instant-segment/src/test_data.rs) to see examples of how you might do that.
```

Comment:
I think the links to the tests would be easier to consume if just had "check out the tests (Python, Rust)", linking from the language names.